src/web/components/elements/VimeoBlockComponent.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import { css } from 'emotion';
+import { Caption } from '@root/src/web/components/Caption';
+
+const responsiveAspectRatio = (height: number, width: number) => css`
+    /* https://css-tricks.com/aspect-ratio-boxes/ */
+    padding-bottom: ${(height / width) * 100}%;
+    position: relative;
+    iframe {
+        width: 100%;
+        height: 100%;
+        position: absolute;
+        top: 0;
+        left: 0;
+    }
+`;
+export const VimeoBlockComponent: React.FC<{
+    pillar: Pillar;
+    url: string;
+    height: number;
+    width: number;
+    caption: string;
+    credit: string;
+    title: string;
+}> = ({ url, caption, title, pillar, width, height }) => {
+    // 812 is the full height on an iphone X
+    const maxHeight = 812;
+    const aspectRatio = width / height;
+    const maxWidth = maxHeight * aspectRatio;
```

Comment:
Could you add a comment for what this this max width is avoiding happening (and why it needs iphone x height)?

Reply:
@gtrufitt It's taken from here, https://github.com/guardian/frontend/pull/17902
We can blame @RichieAHB

Reply:
```suggestion
	// Constrain iframe embeds with a width to their natural width 
	// rather than stretch them to the container using
	// a max that would prevent portrait videos from being taller than an iphone X (baseline)
	// More context: https://github.com/guardian/frontend/pull/17902
    const maxWidth = maxHeight * aspectRatio;
```